src/__tests__/app/preview-nhs-app-template/page.test.tsx

```diff
@@ -1,73 +1,94 @@
-  redirect: () => {
-    throw new Error('Simulated redirect');
-  },
-  useRouter: () => {},
+describe('PreviewNhsAppTemplatePage', () => {
+  beforeEach(jest.resetAllMocks);
 
-  RedirectType: {
-    push: 'push',
-    replace: 'replace',
-  },
-}));
+  it('should load page', async () => {
+    getSessionMock.mockResolvedValueOnce({
+      id: 'session-id',
+      templateType: TemplateType.NHS_APP,
+      nhsAppTemplateName: 'template-name',
+      nhsAppTemplateMessage: 'template-message',
+    });
 
-jest.mock('react-dom', () => {
-  const originalModule = jest.requireActual('react-dom');
-
-  return {
-    ...originalModule,
-    useFormState: (
-      _: (
-        formState: TemplateFormState,
-        formData: FormData
-      ) => Promise<TemplateFormState>,
-      initialState: TemplateFormState
-    ) => [initialState, '/action'],
-  };
-});
-
-test('PreviewNhsAppTemplatePage', async () => {
-  mockSessionSupplier.mockSession = {
-    id: 'session-id',
-    templateType: TemplateType.NHS_APP,
-    nhsAppTemplateName: 'template-name',
-    nhsAppTemplateMessage: 'template-message',
-  };
+    const page = await PreviewNhsAppTemplatePage({
+      params: {
+        sessionId: 'session-id',
+      },
+    });
 
-  const page = await PreviewNhsAppTemplatePage({
-    params: {
-      sessionId: 'session-id',
-    },
+    expect(page).toMatchSnapshot();
   });
 
-  const container = render(page);
-
-  expect(container.asFragment()).toMatchSnapshot();
-});
-
-test('PreviewNhsAppTemplatePage - should handle invalid session', async () => {
-  mockSessionSupplier.mockSession = undefined;
-  const redirectSpy = jest.spyOn(nav, 'redirect');
-
-  await expect(
-    PreviewNhsAppTemplatePage({
+  it('should redirect to invalid-session when no session is found', async () => {
+    await PreviewNhsAppTemplatePage({
       params: {
         sessionId: 'session-id',
       },
-    })
-  ).rejects.toThrow('Simulated redirect');
+    });
+
+    expect(redirectMock).toHaveBeenCalledWith('/invalid-session', 'replace');
+  });
+
+  test.each([
+    {
+      templateType: TemplateType.LETTER,
+      nhsAppTemplateName: 'valid-name',
+      nhsAppTemplateMessage: 'valid-message',
+    },
+    {
+      templateType: TemplateType.EMAIL,
+      nhsAppTemplateName: 'valid-name',
+      nhsAppTemplateMessage: 'valid-message',
+    },
+    {
+      templateType: TemplateType.SMS,
+      nhsAppTemplateName: 'valid-name',
+      nhsAppTemplateMessage: 'valid-message',
+    },
+    {
+      templateType: TemplateType.NHS_APP,
+      nhsAppTemplateName: 'name-1',
+      // Note: We have need this casting because nhsAppTemplateMessage on Session type is required
+      nhsAppTemplateMessage: undefined as unknown as string,
+    },
+    {
+      templateType: TemplateType.NHS_APP,
+      // Note: We have need this casting because nhsAppTemplateName on Session type is required
+      nhsAppTemplateName: undefined as unknown as string,
```

Comment:
I appreciate the comments and I have no solution. But is there a way to somehow explain this via code a new type or something?

Reply:
I was thinking about jigging the typings slightly. But since we're moving to an API solution in the near future, I've left the typings as they are and hopefully when we build the API we refactoring the typing.